e2e/keywords/workload.resource

```diff
@@ -21,6 +41,15 @@ Create statefulset ${idx} with ${volume_type} and ${option} volume
     ${statefulset_name} =    create_statefulset    ${volume_type}    ${option}
     Insert Into List    ${statefulset_list}    ${idx}    ${statefulset_name}
 
+Expand statefulset ${idx} volume by ${size} MiB
```

Comment:
Since we've been able to get pvc name from workload name, maybe we can hide the details in the lower layer like `keywords folder` or even `workload.py` to keep it cohesive on a best effort and save the variable you need to record in resource layer and reduce resource layer complexity.

Reply:
```
${expected_size} = expand_pvc_size_by_mib ${persistentvolumeclaim_list}[${idx}] ${size}
```
And
```
${pvc_name} = get_workload_pvc_name ${statefulset_list}[${idx}]
${expected_size} = expand_pvc_size_by_mib ${pvc_name} ${size}
```
Can you help to elaborate more on why the former is more transparent than the latter?

Reply:
I am confused. Why change to the latter? Is it about the `${persistentvolumeclaim_list}`? We cannot get the workload PVC when StatefulSet are scaled down.

Reply:
Yes, workload scaled down can be a reason. 

But again, for a test case involves multiple different type of workloads, like the following existing test case, how do we distinguish one from others using a single `${persistentvolumeclaim_list}`?

```
Given Create deployment 0 with rwo volume
And Create deployment 1 with rwx volume
And Create deployment 2 with rwo and strict-local volume
And Create statefulset 0 with rwo volume
And Create statefulset 1 with rwx volume
And Create statefulset 2 with rwo and strict-local volume
```

Reply:
1. What scenario are we testing that would need different types of workloads?
1. Currently, the list of persistent volume claims (and other created lists) is differentiated by index. For your case and the way the current codebase implemented, we use the index to distinguish them:
    ```robot
    Given Create deployment 0 with rwo volume
    And Create deployment 1 with rwx volume
    And Create deployment 2 with rwo and strict-local volume
    And Create statefulset 3 with rwo volume
    And Create statefulset 4 with rwx volume
    And Create statefulset 5 with rwo and strict-local volume
    ```
    But to be honest, I don't think using an index is the best way to go about it. It would probably be more logical and clearer if we distinguished them by the name of the workload instead. And the improvement we are discussing should be done in a separate PR.
1. Introducing of the `${persistentvolumeclaim_list}` is to handle the PVC of the downscaled statefulset.

Reply:
> I don't think using an index is the best way to go about it. It would probably be more logical and clearer if we distinguished them by the name of the workload instead.

Yes, totally agree.

Reply:
Lets handle in https://github.com/longhorn/longhorn/issues/7034.

Reply:
> What scenario are we testing that would need different types of workloads?

It's just an example not just for `pvc - workload mapping` but also can be generalize to other situation like the `write data/snapshot/backup - checksum mapping` [below](https://github.com/longhorn/longhorn-tests/pull/1577#discussion_r1379931280). They're actually the same problem.

But for the original question, making the test framework flexible to the maximum helps us to meet any requirements. It makes us be able to randomly combine steps to mimic users' environment/setup and try to reproduce their issue.

Reply:
> We cannot get the workload PVC when StatefulSet is scaled down.

How about `get pvc from deployment/statefulset` instead of `get pvc from pod`. If it uses `volumes.persistentVolumeClaim.claimName`, the pvc name has already been defined. If it uses `volumeClaimTemplates`, the pvc name follows the convention of `volumeClaimTemplates.metadata.name` + `StatefulSet name` + `index`. 

Does this solve the problem when StatefulSet is scaled down?

Reply:
I prefer not to complicate this further as the `${persistentvolumeclaim_list}` works for this PR. Will handle the refactoring in https://github.com/longhorn/longhorn/issues/7034.